gobblin-compaction/src/main/java/gobblin/compaction/mapreduce/avro/AvroDeltaFieldNameProvider.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright (C) 2014-2016 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package gobblin.compaction.mapreduce.avro;
+
+import java.util.List;
+import java.util.concurrent.ExecutionException;
+
+import org.apache.avro.Schema;
+import org.apache.avro.generic.GenericRecord;
+import org.apache.hadoop.conf.Configuration;
+
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.CacheLoader;
+import com.google.common.cache.LoadingCache;
+
+
+/**
+ * Provides the delta field names for {@link AvroKeyDedupReducer}.
+ */
+public abstract class AvroDeltaFieldNameProvider {
+  private final LoadingCache<Schema, List<String>> recordSchemaToDeltaSchemaCache;
+
+  public AvroDeltaFieldNameProvider(Configuration conf) {
+    this.recordSchemaToDeltaSchemaCache=
+        CacheBuilder.newBuilder().maximumSize(100).build(new CacheLoader<Schema, List<String>>() {
+          @Override
+          public List<String> load(Schema schema)
+              throws Exception {
+            return getDeltaFieldNamesForNewSchema(schema);
+          }
+        });
+  }
+
+  public abstract List<String> getDeltaFieldNamesForNewSchema(Schema originalSchema);
+
+  public List<String> getDeltaFieldNames(GenericRecord record) {
+    try {
+      return recordSchemaToDeltaSchemaCache.get(record.getSchema());
```

Comment:
Made changes to the `ConfBasedDeltaFieldProvider` so that it simply returns the field from job config without checking the record schema. I think it achieves the same effect as a simple comparator.